streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1522,6 +1525,92 @@ private void maybeScheduleFollowupRebalance(final long encodedNextScheduledRebal
         }
     }
 
+    private AssignmentError validateTaskAssignment(final ApplicationState applicationState,
+                                                   final TaskAssignment taskAssignment) {
+        final Collection<KafkaStreamsAssignment> assignments = taskAssignment.assignment();
+
+        // Check for AssignmentError.ACTIVE_TASK_ASSIGNED_MULTIPLE_TIMES
+        final Set<TaskId> activeTasks = new HashSet<>();
+        for (final KafkaStreamsAssignment assignment : assignments) {
+            for (final KafkaStreamsAssignment.AssignedTask task : assignment.assignment()) {
+                if (task.type() != KafkaStreamsAssignment.AssignedTask.Type.ACTIVE) {
+                    continue;
+                }
+
+                if (activeTasks.contains(task.id())) {
+                    return AssignmentError.ACTIVE_TASK_ASSIGNED_MULTIPLE_TIMES;
```

Comment:
we should log an error message any time we find an error before returning from this method, so we can include the offending taskId/processId/etc in the log message while we still have that info in scope. 

so ditto for everywhere else we return an error code